### PR TITLE
D3ASIM-3386: Added area name validator

### PR DESCRIPTION
**Pull request by Hassan754**

### CODE REVIEW

**Review by faizan2590 — DISMISSED** [review]
LGTM, please get the area name validated at the d3a side too.

File: d3a_interface/scenario_validators.py [hunk 1]
```diff
@@ -17,8 +17,19 @@
...
 def scenario_validator(scenario_repr):
     validate(instance=scenario_repr, schema=ScenarioSchemas.scenario_schema)
+
+
+def validate_area_name(area_dict: dict):
+    area_name = area_dict.get('name', '')
+    intersect = set(area_name).intersection(
+        ConstSettings.GeneralSettings.AREA_NAME_RESTRICTED_CHARS)
+    if intersect:
+        raise D3AAreaException('Area name cannot have special characters {intersect}.')
```
Comment (hannesdiedrich) [inline d3a_interface/scenario_validators.py:35]:
```suggestion
        raise D3AAreaException(f'Area name cannot have special characters {intersect}.')
```

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by faizan2590 — APPROVED** [review]
LGTM

**Pull request merged**